schema-lending-extended.graphql

```diff
@@ -0,0 +1,1173 @@
+
+  " Token that need to be deposited to take a position in protocol "
+  inputToken: Token!
+
+  " Token that is minted to track ownership of position in protocol (e.g. aToken, cToken). Leave as null if doesn't exist "
+  outputToken: Token
+
+  " Aditional tokens that are given as reward for position in a protocol, usually in liquidity mining programs. e.g. SUSHI in the Onsen program, MATIC for Aave Polygon "
+  rewardTokens: [RewardToken!]
+
+  ##### Quantitative Data #####
+
+  " All interest rates / fees allowed in the market. Interest rate should be in APY percentage "
+  rates: [InterestRate!]!
+
+  " Current TVL (Total Value Locked) of this market "
+  totalValueLockedUSD: BigDecimal!
+
+  " All revenue generated by the market, accrued to the supply side. "
+  cumulativeSupplySideRevenueUSD: BigDecimal!
+
+  " All revenue generated by the market, accrued to the protocol. "
+  cumulativeProtocolSideRevenueUSD: BigDecimal!
+
+  " All revenue generated by the market. "
+  cumulativeTotalRevenueUSD: BigDecimal!
+
+  " Current balance of all deposited assets (not historical cumulative), in USD "
+  totalDepositBalanceUSD: BigDecimal!
+
+  " Sum of all historical deposits in USD (only considers deposits and not withdrawals) "
+  cumulativeDepositUSD: BigDecimal!
+
+  " Current balance of all borrowed/minted assets (not historical cumulative), in USD "
+  totalBorrowBalanceUSD: BigDecimal!
+
+  " Sum of all historical borrows/mints in USD (i.e. total loan origination) "
+  cumulativeBorrowUSD: BigDecimal!
+
+  " Sum of all historical liquidations in USD "
+  cumulativeLiquidateUSD: BigDecimal!
+
+  ##### Token Balances #####
+
+  " Amount of input tokens in the market "
+  inputTokenBalance: BigInt!
+
+  " Price per share of input token in USD "
+  inputTokenPriceUSD: BigDecimal!
+
+  " Total supply of output token "
+  outputTokenSupply: BigInt!
+
+  " Price per share of output token in USD "
+  outputTokenPriceUSD: BigDecimal!
+
+  " Amount of input token per full share of output token. Only applies when the output token exists (note this is a ratio and not a percentage value, i.e. 1.05 instead of 105%) "
+  exchangeRate: BigDecimal
+
+  " Per-block reward token emission as of the current block normalized to a day, in token's native amount. This should be ideally calculated as the theoretical rate instead of the realized amount. "
+  rewardTokenEmissionsAmount: [BigInt!]
+
+  " Per-block reward token emission as of the current block normalized to a day, in USD value. This should be ideally calculated as the theoretical rate instead of the realized amount. "
+  rewardTokenEmissionsUSD: [BigDecimal!]
+
+  " Creation timestamp "
+  createdTimestamp: BigInt!
+
+  " Creation block number "
+  createdBlockNumber: BigInt!
+
+  ##### Account/Position Data #####
+
+  " All open positions in this market "
+  positions: [Positions!]! @derivedFrom(field: "market")
+
+  " Number of open positions in this market "
+  positionCount: Int!
```

Comment:
Maybe overkill: It may be worth it to break this up into deposit & borrow position counts

Reply:
What would be some use cases for this? There are some performance reasons for this design but would want to weight the trade-off

Reply:
It's another way to show long/short bias. Instead of long/short bias in USD, now its long/short Bias in terms of users.